### PR TITLE
Facet refactoring & DateRangeFacet addition

### PR DESCRIPTION
**Pull request by Akrion**
- Refactor the Facet type and extract the common components/functions to `util/facet.js`
- Add DateRangeFacet
- Allow for the hiding of the radio list and select all for facets

image

in regards to https://github.com/smartprocure/spark/issues/6298

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 62c77fc8e428c82507dadc65f08150a5121a7cdc


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 205c55235a9729b5c59546afde5abc503ca09108
</p>

File: src/exampleTypes/Facet.js [hunk 1]
```diff
@@ -115,13 +117,15 @@ let Facet = ({
   theme: { Checkbox, RadioList },
 }) => (
   <div className="contexture-facet">
-    <RadioList
-      value={node.mode || 'include'} // Fix by changing defaults in client example type
-      onChange={mode => tree.mutate(node.path, { mode })}
-      options={F.autoLabelOptions(['include', 'exclude'])}
-    />
+    {!hide.facetFilter && (
```
Comment (stellarhoof) [inline src/exampleTypes/Facet.js:120]: This should be `hide.radioList`

Comment (Akrion) [reply src/exampleTypes/Facet.js]: Fixed

**Comment by Akrion** [comment]
@daedalus28 @stellarhoof Facet is now refactored (common stuff with DateRangeFacet extracted). DateRangeFacet component added. Need this merged to close the Spark stuff.

File: src/exampleTypes/DateRangeFacet.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  tree,
+  node,
+  hide = {
+    counts: false, // Hide the facet counts so only the labels are displayed
+  },
+  display = displayFn,
+  displayBlank = displayBlankFn,
+  formatCount = toNumber,
+}) => (
+  <div className="contexture-facet">
```
Comment (daedalus28) [inline src/exampleTypes/DateRangeFacet.js:16]: we should change this to `contexture-daterangefacet`

Comment (Akrion) [reply src/exampleTypes/DateRangeFacet.js]: done.

**Pull request merged**